src/helpers/Monitor.cpp

```diff
@@ -682,6 +682,10 @@ void CMonitor::setSpecialWorkspace(const PHLWORKSPACE& pWorkspace) {
     }
 
     if (activeSpecialWorkspace) {
+        // do not do anything if it is already open
+        if (activeSpecialWorkspace == pWorkspace)
```

Comment:
why do the check here and not at the top

Reply:
just put it where the workspace was unset, which is what causes the problem
but yeah, up top makes more sense - can also save one whole monitor damage